build.rs

```diff
@@ -0,0 +1,9 @@
+fn main() {
+    // Add a custom cfg() to avoid repeated occurrences of
+    // any(feature = "std", feature = "alloc)
+    if cfg!(any(feature = "std", feature = "alloc")) {
+        println!("cargo::rustc-cfg=HAVE_ALLOCATION");
+    }
+    println!("cargo::rustc-check-cfg=cfg(HAVE_ALLOCATION)");
+    println!("cargo::rerun-if-changed=build.rs");
+}
```

Comment:
hmm I'm not 100% sure adding a `build.rs` is worth it. IMO `any(feature = "alloc", feature = "std")` is not that bad. I'd prefer if we have no `build.rs` at all.

Reply:
Yes, without feature = "arc" polluting everything it's ok.